### PR TITLE
When aborting a merge/cherry-pick, avoid resetting ignored tables.

### PR DESCRIPTION
**Pull request by nicktobey**
Tables that match `dolt_ignore` in the working set should be ignored by most operations.

(CAVEAT: If we want to match git, these tables should only be ignored if there are not staged/HEAD versions of those tables. But we appear to not match git in that regard right now.)

This PR changes the logic for aborting merges to avoid modifying any tables match `dolt_ignore`

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>826d9fc</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>826d9fc</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>826d9fc</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

**Comment by nicktobey** [comment]
It appears that prior to this PR, we were leaving all tables staged after an aborted merge, except for ignored tables. This seems incorrect: we should not be leaving any tables staged after aborting a merge, given that you shouldn't be able to start a merge while there are staged tables, and aborting the merge should return the user to the state they were in before the merge began.

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>1f12466</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>1f12466</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>1f12466</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go [hunk 1]
```diff
@@ -5709,9 +5723,22 @@ var DoltCherryPickTests = []queries.ScriptTest{
 				Expected: []sql.Row{{1, "one"}},
 			},
 			{
+				// An ignored table should still be present (and unstaged) after aborting the merge.
 				Query:    "select * from dolt_status;",
-				Expected: []sql.Row{},
+				Expected: []sql.Row{{"generated_foo", false, "new table"}},
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/enginetest/dolt_queries.go:5728]: Maybe beyond the scope of this bug fix, but I would expect that an ignored table doesn't appear in `dolt_status`, similar to how ignored files don't appear in `git status`.

**Review by zachmu — APPROVED** [review]
LG, but see comment about behavior of dolt_status

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go (hunk 1 above)
Comment (nicktobey) [reply go/libraries/doltcore/sqle/enginetest/dolt_queries.go:5728]: I think when `dolt_ignore` was being added, the decision was to add a `ignored` column to the status table.

Either way, I agree that something should be done, and also agree that this is beyond the scope of the bug fix.

**Pull request merged**